scripts/annotation-am-migration-main.sql

```diff
@@ -0,0 +1,116 @@
+
+BEGIN;
+
+DROP TABLE IF EXISTS access_management_migration_errors;
+CREATE TABLE access_management_migration_errors (
+    document_id	VARCHAR,
+    accessor_type ACCESSOR_TYPE,
+    user_id VARCHAR,
+    annotation_id VARCHAR,
+    permissions VARCHAR,
+    resource_name VARCHAR
+);
+
+CREATE TEMP TABLE stage (
+    document_id	VARCHAR,
+    accessor_type ACCESSOR_TYPE,
+    user_id VARCHAR,
+    annotation_id VARCHAR,
+    permissions VARCHAR,
+    resource_name VARCHAR
+);
+
+ALTER TABLE access_management DROP CONSTRAINT access_management_unique;
+ALTER TABLE access_management DROP CONSTRAINT access_management_resources_fkey;
+ALTER TABLE access_management DROP CONSTRAINT relationship_fkey;
+
+\COPY stage FROM 'am-migration.csv' DELIMITER ',' CSV HEADER;
+
+ALTER TABLE stage ADD COLUMN permissions_int INTEGER DEFAULT 0;
+UPDATE stage SET permissions_int = permissions_int + 1
+WHERE permissions LIKE '%CREATE%';
+UPDATE stage SET permissions_int = permissions_int + 2
+WHERE permissions LIKE '%READ%';
+UPDATE stage SET permissions_int = permissions_int + 4
+WHERE permissions LIKE '%UPDATE%';
+UPDATE stage SET permissions_int = permissions_int + 8
+WHERE permissions LIKE '%DELETE%';
+
+SELECT COUNT(*) AS "rows to migrate" FROM stage;
+
+WITH file_duplicates AS (
+    DELETE FROM stage a USING stage b
+    WHERE a.ctid < b.ctid
+        AND a.document_id = b.document_id
+        AND a.accessor_type = b.accessor_type
+        AND a.user_id = b.user_id
+        AND a.annotation_id = b.annotation_id
+        AND a.permissions = b.permissions
+        AND a.resource_name = b.resource_name
+    RETURNING *
+)
+SELECT COUNT(*) AS "duplicate rows in file (skipping)" FROM file_duplicates;
+
+WITH access_management_duplicates AS (
+    DELETE FROM stage a USING access_management b
+    WHERE a.document_id = b.resource_id
+        AND a.accessor_type = b.accessor_type
+        AND a.user_id = b.accessor_id
+        AND a.annotation_id = b."attribute"
+        AND a.permissions_int = b.permissions
+        AND a.resource_name = b.resource_name
+    RETURNING *
```

Comment:
include resource_type in the where clause

Reply:
👍 well spotted